overlord/snapstate/handlers.go

```diff
@@ -977,15 +977,23 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func snapdSnapInstalled(st *state.State) bool {
+	var snapst SnapState
+	err := Get(st, "snapd", &snapst)
+	return err == nil && snapst.IsInstalled()
```

Comment:
err == nil is enough ,   IsInstalled is only significant in context where we ignore ErrNoState